Broken windows build re-enable

Windows build broke due to some external change.

The following warnings were emitted during compilation:

warning: gcc.exe: warning: /std:c11: linker input file unused because linking not done
warning: gcc.exe: error: /std:c11: linker input file not found: No such file or directory

error: failed to run custom build command for `c-kzg v0.1.1`

Caused by:
  process didn't exit successfully: `D:\a\redgold\redgold\target\release\build\c-kzg-834b9980e88d8be5\build-script-build` (exit code: 101)
  --- stdout
  TARGET = Some("x86_64-pc-windows-gnu")
  OPT_LEVEL = Some("1")
  HOST = Some("x86_64-pc-windows-gnu")
  cargo:rerun-if-env-changed=CC_x86_64-pc-windows-gnu
  CC_x86_64-pc-windows-gnu = None
  cargo:rerun-if-env-changed=CC_x86_64_pc_windows_gnu
  CC_x86_64_pc_windows_gnu = None
  cargo:rerun-if-env-changed=HOST_CC
  HOST_CC = None
  cargo:rerun-if-env-changed=CC
  CC = None
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("false")
  CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
  cargo:rerun-if-env-changed=CFLAGS_x86_64-pc-windows-gnu
  CFLAGS_x86_64-pc-windows-gnu = None
  cargo:rerun-if-env-changed=CFLAGS_x86_64_pc_windows_gnu
  CFLAGS_x86_64_pc_windows_gnu = None
  cargo:rerun-if-env-changed=HOST_CFLAGS
  HOST_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = None
  running: "gcc.exe" "-O1" "-ffunction-sections" "-fdata-sections" "-m64" "-I" "C:\\Users\\runneradmin\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\c-kzg-0.1.1\\c-kzg-sys\\blst\\bindings" "-D_CRT_SECURE_NO_WARNINGS" "/std:c11" "-DLIB_PREFIX=ckzg" "-DFIELD_ELEMENTS_PER_BLOB=4096" "-o" "D:\\a\\redgold\\redgold\\target\\x86_64-pc-windows-gnu\\release\\build\\c-kzg-74d4b4805e1e6964\\out\\c08e3e867c0ecc26-c_kzg_4844.o" "-c" "C:\\Users\\runneradmin\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\c-kzg-0.1.1\\c-kzg-sys\\src\\c_kzg_4844.c"
  cargo:warning=gcc.exe: warning: /std:c11: linker input file unused because linking not done

  cargo:warning=gcc.exe: error: /std:c11: linker input file not found: No such file or directory

  exit code: 1

  --- stderr
  Using LIB_PREFIX=ckzg
  Using FIELD_ELEMENTS_PER_BLOB=4096
  thread 'main' panicked at 'Failed to compile ckzg: Error { kind: ToolExecError, message: "Command \"gcc.exe\" \"-O1\" \"-ffunction-sections\" \"-fdata-sections\" \"-m64\" \"-I\" \"C:\\\\Users\\\\runneradmin\\\\.cargo\\\\registry\\\\src\\\\index.crates.io-6f17d22bba15001f\\\\c-kzg-0.1.1\\\\c-kzg-sys\\\\blst\\\\bindings\" \"-D_CRT_SECURE_NO_WARNINGS\" \"/std:c11\" \"-DLIB_PREFIX=ckzg\" \"-DFIELD_ELEMENTS_PER_BLOB=4096\" \"-o\" \"D:\\\\a\\\\redgold\\\\redgold\\\\target\\\\x86_64-pc-windows-gnu\\\\release\\\\build\\\\c-kzg-74d4b4805e1e6964\\\\out\\\\c08e3e867c0ecc26-c_kzg_4844.o\" \"-c\" \"C:\\\\Users\\\\runneradmin\\\\.cargo\\\\registry\\\\src\\\\index.crates.io-6f17d22bba15001f\\\\c-kzg-0.1.1\\\\c-kzg-sys\\\\src\\\\c_kzg_4844.c\" with args \"gcc.exe\" did not execute successfully (status code exit code: 1)." }', C:\Users\runneradmin\.cargo\registry\src\index.crates.io-6f17d22bba15001f\c-kzg-0.1.1\build.rs:52:30
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
Error: Process completed with exit code 101.


  build-windows:
    #    needs: [test]
    if:
      contains('
      refs/heads/predev
      refs/heads/dev
      refs/heads/staging
      refs/heads/test
      refs/heads/main
      ', github.ref)
    runs-on: windows-2019
    timeout-minutes: 100
    permissions:
      contents: write
    env:
      DATABASE_URL: "sqlite:///Users/runner/work/redgold-core/redgold-core/.rg/sqlx/data_store.sqlite"
    outputs:
      release_match: ${{ steps.check-branch.outputs.match }}
    steps:
      - name: Checkout
        uses: actions/checkout@v1
      - name: Install latest rust toolchain
        uses: actions-rs/toolchain@v1
        with:
          toolchain: ${{ env.TOOLCHAIN }}
          target: x86_64-pc-windows-gnu
          default: true
          override: true
      - uses: msys2/setup-msys2@v2
      - run: echo "asdf" > ./redgold.exe
      - shell: msys2 {0}
        name: "Msys2 cargo build with dependencies"
        if:
          contains('
          refs/heads/staging
          refs/heads/test
          refs/heads/main
          ', github.ref)
        run: |
          export PATH="$HOME/.cargo/bin:$PATH"
          uname -a
          ls -lA
          pwd
          pacman -S --noconfirm mingw-w64-x86_64-gmp
          pacman -S --noconfirm mingw-w64-x86_64-rust
          pacman -S --noconfirm mingw-w64-x86_64-openssl
          pacman -S --noconfirm make 
          echo "OPENSSL_DIR=/mingw64" >> $GITHUB_ENV
          echo "OPENSSL_STATIC=yes" >> $GITHUB_ENV
          cargo build --release --target x86_64-pc-windows-gnu
          ls 
          ls target/
          ls target/x86_64-pc-windows-gnu/
          ls target/x86_64-pc-windows-gnu/release/
          cp target/x86_64-pc-windows-gnu/release/redgold.exe ./redgold.exe
      - uses: actions/upload-artifact@v3
        with:
          name: redgold.exe
          path: ./redgold.exe

ChatGPT

The primary issue you're facing is that the gcc compiler is being passed a /std:c11 flag, which is not a valid flag for gcc. This flag is typically used with Microsoft's cl compiler.

To address this:

    Check the build.rs of the c-kzg crate you are compiling. It seems that /std:c11 is being passed as a compiler flag. It may need to be changed or removed.
    If c-kzg crate is a third-party crate and you don't have control over its build.rs, you might need to either:
        Report an issue to the maintainers of the crate, asking for better Windows support.
        Fork the crate and modify it to suit your needs.

Considering your GitHub Actions script, it seems you are using a mixture of native Windows tools and MSYS2. Be careful with this as mixing these environments can cause issues. You might need to ensure that you're using the right gcc (from MSYS2, not any native Windows one).

Here's a possible action plan:

    Ensure you're using the right gcc.
        After setting up MSYS2, add its bin directory to the PATH. For example, add:

        yaml

        echo "/mingw64/bin:$PATH" >> $GITHUB_ENV

        after your pacman commands.
    Modify or ensure the build scripts of the crate don't include flags incompatible with gcc.
    If you're overriding the compiler or providing flags globally, ensure they're compatible with the crates you're building.

Lastly, ensure that you've set the TOOLCHAIN environment variable, as it's being used in the GitHub Actions but isn't defined in the provided snippet.

--- a/.cargo/config.toml
+++ b/.cargo/config.toml
@@ -1,0 +1,2 @@
+[target.x86_64-pc-windows-gnu]
+linker = "x86_64-w64-mingw32-gcc"

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -224,12 +224,12 @@ jobs:
          Currently broken due to external change.
       - shell: msys2 {0}
         name: "Msys2 cargo build with dependencies"
-        if:
-          contains('
-          refs/heads/staging
-          refs/heads/test
-          refs/heads/main
-          ', github.ref)
+#        if:
+#          contains('
+#          refs/heads/staging
+#          refs/heads/test
+#          refs/heads/main
+#          ', github.ref)
         run: |
           export PATH="$HOME/.cargo/bin:$PATH"
           uname -a

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -276,12 +276,6 @@ jobs:
           pacman -Sy --noconfirm mingw-w64-x86_64-toolchain || true
       - shell: msys2 {0}
         name: "Msys2 cargo build with dependencies"
-        #        if:
-        #          contains('
-        #          refs/heads/staging
-        #          refs/heads/test
-        #          refs/heads/main
-        #          ', github.ref)
         run: |
           export PATH="$HOME/.cargo/bin:$PATH"
           uname -a
@@ -290,12 +284,6 @@ jobs:
           pacman -S --noconfirm mingw-w64-x86_64-rust
           pacman -S --noconfirm mingw-w64-x86_64-openssl
           pacman -S --noconfirm make
-#          echo "OPENSSL_DIR=/mingw64" >> $GITHUB_ENV
-#          echo "OPENSSL_STATIC=yes" >> $GITHUB_ENV
-#          echo "CC=gcc" >> $GITHUB_ENV      CC: gcc
-#          echo "CFLAGS=-std=cli" >> $GITHUB_ENV
-#          export CC=gcc
-#          export CFLAGS="-std=c11"
           cargo build --release --target x86_64-pc-windows-gnu
           ls
           ls target/

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -246,6 +246,8 @@ jobs:
           echo "OPENSSL_STATIC=yes" >> $GITHUB_ENV
           echo "CC=gcc" >> $GITHUB_ENV      CC: gcc
           echo "CFLAGS=-std=cli" >> $GITHUB_ENV
+          export CC=gcc
+          export CFLAGS="-std=c11"
           cargo build --release --target x86_64-pc-windows-gnu
           ls
           ls target/

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -237,6 +237,7 @@ jobs:
           uname -a
           ls -lA
           pwd
+          pacman -Syu --noconfirm
           pacman -S --noconfirm mingw-w64-x86_64-gmp
           pacman -S --noconfirm mingw-w64-x86_64-rust
           pacman -S --noconfirm mingw-w64-x86_64-openssl

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -207,6 +207,8 @@ jobs:
       contents: write
     env:
       DATABASE_URL: "sqlite:///Users/runner/work/redgold-core/redgold-core/.rg/sqlx/data_store.sqlite"
+      CC: gcc
+      CFLAGS: "-std=c11"
     outputs:
       release_match: ${{ steps.check-branch.outputs.match }}
     steps:
@@ -241,6 +243,8 @@ jobs:
           pacman -S --noconfirm make
           echo "OPENSSL_DIR=/mingw64" >> $GITHUB_ENV
           echo "OPENSSL_STATIC=yes" >> $GITHUB_ENV
+          echo "CC=gcc" >> $GITHUB_ENV      CC: gcc
+          echo "CFLAGS=-std=cli" >> $GITHUB_ENV
           cargo build --release --target x86_64-pc-windows-gnu
           ls
           ls target/

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -16,22 +16,21 @@ on:
     types: [opened, synchronize, reopened]
 jobs:
   test:
-    if: false
-#      >
-#      (
-#        github.event_name == 'push' &&
-#        !contains('
-#          refs/heads/predev
-#          refs/heads/dev
-#          refs/heads/staging
-#          refs/heads/test
-#          refs/heads/main
-#        ', github.ref)
-#      ) ||
-#      (
-#        github.event_name == 'pull_request_target' &&
-#        github.event.pull_request.head.repo.full_name != github.repository
-#      )
+    if: >
+      (
+        github.event_name == 'push' && 
+        !contains('
+          refs/heads/predev
+          refs/heads/dev
+          refs/heads/staging
+          refs/heads/test
+          refs/heads/main
+        ', github.ref)
+      ) || 
+      (
+        github.event_name == 'pull_request_target' && 
+        github.event.pull_request.head.repo.full_name != github.repository
+      )
     runs-on: ubuntu-latest-m
     timeout-minutes: 60
     permissions:
@@ -162,49 +161,50 @@ jobs:
         with:
           name: redgold_linux
           path: ./redgold_linux
-  build-linux-windows:
-#    if: ${{ github.event_name != 'pull_request_target' && contains('
-#      refs/heads/predev
-#      refs/heads/dev
-#      refs/heads/staging
-#      refs/heads/test
-#      refs/heads/main
-#      ', github.ref) }}
-    runs-on: ubuntu-latest-m
-    timeout-minutes: 100
-    permissions:
-      contents: write
-#    env:
-#      DATABASE_URL: "sqlite:///home/runner/work/redgold-core/redgold-core/.rg/sqlx/data_store.sqlite"
-    steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v2
-        with:
-          node-version: '14'
-      - uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: ${{ env.TOOLCHAIN }}
-      - uses: Swatinem/rust-cache@v1
-        with:
-          cache-on-failure: true
-        # TODO: Update to use release plugin
-        #      - run: cargo install cargo-release
-        #      - run: cargo release # --execute --no-confirm --token ${{ secrets.GITHUB_TOKEN }}
-      - run: echo "BRANCH=${GITHUB_REF##*/}" >> $GITHUB_ENV
-      - run: echo "Github branch = ${{ env.BRANCH }}"
-      - run: export VERSION=$(head Cargo.toml | grep 'version = ' | cut -d "=" -f 2 | tr -d ' "'); echo "VERSION=$VERSION" >> $GITHUB_ENV
-      - run: echo "Testing version = ${{ env.VERSION }}"
-      - run: ./bin/install_build_deps.sh
-      - run: sudo apt update; sudo apt install -y mingw-w64 mingw-w64-tools
-      - run: rustup target add x86_64-pc-windows-gnu
-      - run: cargo build --target x86_64-pc-windows-gnu --release
-      - run: ls ./target/release -lA
-
-      - uses: actions/upload-artifact@v3
-        with:
-          name: redgold_win_cross
-          path: ./target/release/redgold.exe
+# Potential Alternative if the other build starts failing again, requires compiling lgmp from source for mingw64
+#  build-linux-windows:
+##    if: ${{ github.event_name != 'pull_request_target' && contains('
+##      refs/heads/predev
+##      refs/heads/dev
+##      refs/heads/staging
+##      refs/heads/test
+##      refs/heads/main
+##      ', github.ref) }}
+#    runs-on: ubuntu-latest-m
+#    timeout-minutes: 100
+#    permissions:
+#      contents: write
+##    env:
+##      DATABASE_URL: "sqlite:///home/runner/work/redgold-core/redgold-core/.rg/sqlx/data_store.sqlite"
+#    steps:
+#      - uses: actions/checkout@v2
+#      - uses: actions/setup-node@v2
+#        with:
+#          node-version: '14'
+#      - uses: actions-rs/toolchain@v1
+#        with:
+#          profile: minimal
+#          toolchain: ${{ env.TOOLCHAIN }}
+#      - uses: Swatinem/rust-cache@v1
+#        with:
+#          cache-on-failure: true
+#        # TODO: Update to use release plugin
+#        #      - run: cargo install cargo-release
+#        #      - run: cargo release # --execute --no-confirm --token ${{ secrets.GITHUB_TOKEN }}
+#      - run: echo "BRANCH=${GITHUB_REF##*/}" >> $GITHUB_ENV
+#      - run: echo "Github branch = ${{ env.BRANCH }}"
+#      - run: export VERSION=$(head Cargo.toml | grep 'version = ' | cut -d "=" -f 2 | tr -d ' "'); echo "VERSION=$VERSION" >> $GITHUB_ENV
+#      - run: echo "Testing version = ${{ env.VERSION }}"
+#      - run: ./bin/install_build_deps.sh
+#      - run: sudo apt update; sudo apt install -y mingw-w64 mingw-w64-tools
+#      - run: rustup target add x86_64-pc-windows-gnu
+#      - run: cargo build --target x86_64-pc-windows-gnu --release
+#      - run: ls ./target/release -lA
+#
+#      - uses: actions/upload-artifact@v3
+#        with:
+#          name: redgold_win_cross
+#          path: ./target/release/redgold.exe
   build-linux-ubuntu-20:
     if: ${{ github.event_name != 'pull_request_target' && contains('
       refs/heads/predev
@@ -235,14 +235,13 @@ jobs:
           name: redgold_linux_ubuntu20
           path: ./redgold_linux_ubuntu20
   build-windows:
-    #    needs: [test]
-#    if: ${{ github.event_name != 'pull_request_target' && contains('
-#      refs/heads/predev
-#      refs/heads/dev
-#      refs/heads/staging
-#      refs/heads/test
-#      refs/heads/main
-#      ', github.ref) }}
+    if: ${{ github.event_name != 'pull_request_target' && contains('
+      refs/heads/predev
+      refs/heads/dev
+      refs/heads/staging
+      refs/heads/test
+      refs/heads/main
+      ', github.ref) }}
     runs-on: windows-latest
     timeout-minutes: 100
     permissions:
@@ -268,6 +267,12 @@ jobs:
          Currently broken due to external change.
       - shell: msys2 {0}
         name: "Msys2 cargo build with upgrade"
+        if:
+          contains('
+          refs/heads/staging
+          refs/heads/test
+          refs/heads/main
+          ', github.ref)
         run: |
           export PATH="$HOME/.cargo/bin:$PATH"
           uname -a
@@ -276,6 +281,12 @@ jobs:
           pacman -Sy --noconfirm mingw-w64-x86_64-toolchain || true
       - shell: msys2 {0}
         name: "Msys2 cargo build with dependencies"
+        if:
+          contains('
+          refs/heads/staging
+          refs/heads/test
+          refs/heads/main
+          ', github.ref)
         run: |
           export PATH="$HOME/.cargo/bin:$PATH"
           uname -a

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -280,6 +280,18 @@ jobs:
           ls -lA
           pwd
           pacman -Syu --noconfirm
+      - shell: msys2 {0}
+        name: "Msys2 cargo build with dependencies"
+        #        if:
+        #          contains('
+        #          refs/heads/staging
+        #          refs/heads/test
+        #          refs/heads/main
+        #          ', github.ref)
+        run: |
+          export PATH="$HOME/.cargo/bin:$PATH"
+          uname -a
+          ls -lA
           pacman -S --noconfirm mingw-w64-x86_64-gmp
           pacman -S --noconfirm mingw-w64-x86_64-rust
           pacman -S --noconfirm mingw-w64-x86_64-openssl

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -162,7 +162,49 @@ jobs:
         with:
           name: redgold_linux
           path: ./redgold_linux
-#    needs: [test]
+  build-linux-windows:
+#    if: ${{ github.event_name != 'pull_request_target' && contains('
+#      refs/heads/predev
+#      refs/heads/dev
+#      refs/heads/staging
+#      refs/heads/test
+#      refs/heads/main
+#      ', github.ref) }}
+    runs-on: ubuntu-latest-m
+    timeout-minutes: 100
+    permissions:
+      contents: write
+#    env:
+#      DATABASE_URL: "sqlite:///home/runner/work/redgold-core/redgold-core/.rg/sqlx/data_store.sqlite"
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ env.TOOLCHAIN }}
+      - uses: Swatinem/rust-cache@v1
+        with:
+          cache-on-failure: true
+        # TODO: Update to use release plugin
+        #      - run: cargo install cargo-release
+        #      - run: cargo release # --execute --no-confirm --token ${{ secrets.GITHUB_TOKEN }}
+      - run: echo "BRANCH=${GITHUB_REF##*/}" >> $GITHUB_ENV
+      - run: echo "Github branch = ${{ env.BRANCH }}"
+      - run: export VERSION=$(head Cargo.toml | grep 'version = ' | cut -d "=" -f 2 | tr -d ' "'); echo "VERSION=$VERSION" >> $GITHUB_ENV
+      - run: echo "Testing version = ${{ env.VERSION }}"
+      - run: ./bin/install_build_deps.sh
+      - run: sudo apt update; sudo apt install -y mingw-w64
+      - run: rustup target add x86_64-pc-windows-gnu
+      - run: cargo build --target x86_64-pc-windows-gnu --release
+      - run: ls ./target/release -lA
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: redgold_win_cross
+          path: ./target/release/redgold.exe
   build-linux-ubuntu-20:
     if: ${{ github.event_name != 'pull_request_target' && contains('
       refs/heads/predev

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -267,19 +267,13 @@ jobs:
       - run: echo "asdf" > ./redgold.exe
          Currently broken due to external change.
       - shell: msys2 {0}
-        name: "Msys2 cargo build with dependencies"
-#        if:
-#          contains('
-#          refs/heads/staging
-#          refs/heads/test
-#          refs/heads/main
-#          ', github.ref)
+        name: "Msys2 cargo build with upgrade"
         run: |
           export PATH="$HOME/.cargo/bin:$PATH"
           uname -a
           ls -lA
           pwd
-          pacman -Syu --noconfirm
+          pacman -Sy --noconfirm mingw-w64-x86_64-toolchain || true
       - shell: msys2 {0}
         name: "Msys2 cargo build with dependencies"
         #        if:

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -196,7 +196,7 @@ jobs:
       - run: export VERSION=$(head Cargo.toml | grep 'version = ' | cut -d "=" -f 2 | tr -d ' "'); echo "VERSION=$VERSION" >> $GITHUB_ENV
       - run: echo "Testing version = ${{ env.VERSION }}"
       - run: ./bin/install_build_deps.sh
-      - run: sudo apt update; sudo apt install -y mingw-w64
+      - run: sudo apt update; sudo apt install -y mingw-w64 mingw-w64-tools
       - run: rustup target add x86_64-pc-windows-gnu
       - run: cargo build --target x86_64-pc-windows-gnu --release
       - run: ls ./target/release -lA

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -16,21 +16,22 @@ on:
     types: [opened, synchronize, reopened]
 jobs:
   test:
-    if: >
-      (
-        github.event_name == 'push' && 
-        !contains('
-          refs/heads/predev
-          refs/heads/dev
-          refs/heads/staging
-          refs/heads/test
-          refs/heads/main
-        ', github.ref)
-      ) || 
-      (
-        github.event_name == 'pull_request_target' && 
-        github.event.pull_request.head.repo.full_name != github.repository
-      )
+    if: false
+#      >
+#      (
+#        github.event_name == 'push' &&
+#        !contains('
+#          refs/heads/predev
+#          refs/heads/dev
+#          refs/heads/staging
+#          refs/heads/test
+#          refs/heads/main
+#        ', github.ref)
+#      ) ||
+#      (
+#        github.event_name == 'pull_request_target' &&
+#        github.event.pull_request.head.repo.full_name != github.repository
+#      )
     runs-on: ubuntu-latest-m
     timeout-minutes: 60
     permissions:
@@ -193,13 +194,13 @@ jobs:
           path: ./redgold_linux_ubuntu20
   build-windows:
     #    needs: [test]
-    if: ${{ github.event_name != 'pull_request_target' && contains('
-      refs/heads/predev
-      refs/heads/dev
-      refs/heads/staging
-      refs/heads/test
-      refs/heads/main
-      ', github.ref) }}
+#    if: ${{ github.event_name != 'pull_request_target' && contains('
+#      refs/heads/predev
+#      refs/heads/dev
+#      refs/heads/staging
+#      refs/heads/test
+#      refs/heads/main
+#      ', github.ref) }}
     runs-on: windows-latest
     timeout-minutes: 100
     permissions:
@@ -220,32 +221,32 @@ jobs:
           override: true
       - uses: msys2/setup-msys2@v2
       - run: echo "asdf" > ./redgold.exe
-        # Currently broken due to external change.
-#      - shell: msys2 {0}
-#        name: "Msys2 cargo build with dependencies"
-#        if:
-#          contains('
-#          refs/heads/staging
-#          refs/heads/test
-#          refs/heads/main
-#          ', github.ref)
-#        run: |
-#          export PATH="$HOME/.cargo/bin:$PATH"
-#          uname -a
-#          ls -lA
-#          pwd
-#          pacman -S --noconfirm mingw-w64-x86_64-gmp
-#          pacman -S --noconfirm mingw-w64-x86_64-rust
-#          pacman -S --noconfirm mingw-w64-x86_64-openssl
-#          pacman -S --noconfirm make
-#          echo "OPENSSL_DIR=/mingw64" >> $GITHUB_ENV
-#          echo "OPENSSL_STATIC=yes" >> $GITHUB_ENV
-#          cargo build --release --target x86_64-pc-windows-gnu
-#          ls
-#          ls target/
-#          ls target/x86_64-pc-windows-gnu/
-#          ls target/x86_64-pc-windows-gnu/release/
-#          cp target/x86_64-pc-windows-gnu/release/redgold.exe ./redgold.exe
+         Currently broken due to external change.
+      - shell: msys2 {0}
+        name: "Msys2 cargo build with dependencies"
+        if:
+          contains('
+          refs/heads/staging
+          refs/heads/test
+          refs/heads/main
+          ', github.ref)
+        run: |
+          export PATH="$HOME/.cargo/bin:$PATH"
+          uname -a
+          ls -lA
+          pwd
+          pacman -S --noconfirm mingw-w64-x86_64-gmp
+          pacman -S --noconfirm mingw-w64-x86_64-rust
+          pacman -S --noconfirm mingw-w64-x86_64-openssl
+          pacman -S --noconfirm make
+          echo "OPENSSL_DIR=/mingw64" >> $GITHUB_ENV
+          echo "OPENSSL_STATIC=yes" >> $GITHUB_ENV
+          cargo build --release --target x86_64-pc-windows-gnu
+          ls
+          ls target/
+          ls target/x86_64-pc-windows-gnu/
+          ls target/x86_64-pc-windows-gnu/release/
+          cp target/x86_64-pc-windows-gnu/release/redgold.exe ./redgold.exe
       - uses: actions/upload-artifact@v3
         with:
           name: redgold.exe

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -249,8 +249,8 @@ jobs:
       contents: write
     env:
       DATABASE_URL: "sqlite:///Users/runner/work/redgold-core/redgold-core/.rg/sqlx/data_store.sqlite"
-      CC: gcc
-      CFLAGS: "-std=c11"
+#      CC: gcc
+#      CFLAGS: "-std=c11"
     outputs:
       release_match: ${{ steps.check-branch.outputs.match }}
     steps:
@@ -290,12 +290,12 @@ jobs:
           pacman -S --noconfirm mingw-w64-x86_64-rust
           pacman -S --noconfirm mingw-w64-x86_64-openssl
           pacman -S --noconfirm make
-          echo "OPENSSL_DIR=/mingw64" >> $GITHUB_ENV
-          echo "OPENSSL_STATIC=yes" >> $GITHUB_ENV
-          echo "CC=gcc" >> $GITHUB_ENV      CC: gcc
-          echo "CFLAGS=-std=cli" >> $GITHUB_ENV
-          export CC=gcc
-          export CFLAGS="-std=c11"
+#          echo "OPENSSL_DIR=/mingw64" >> $GITHUB_ENV
+#          echo "OPENSSL_STATIC=yes" >> $GITHUB_ENV
+#          echo "CC=gcc" >> $GITHUB_ENV      CC: gcc
+#          echo "CFLAGS=-std=cli" >> $GITHUB_ENV
+#          export CC=gcc
+#          export CFLAGS="-std=c11"
           cargo build --release --target x86_64-pc-windows-gnu
           ls
           ls target/

--- a/executor/Cargo.toml
+++ b/executor/Cargo.toml
@@ -43,7 +43,7 @@ parity-scale-codec = "3.6.4"
 
 
 # REVM related
-revm = {version = "3.3.0", features = ["ethersdb", "dev"]}
+#revm = {version = "3.3.0", features = ["ethersdb", "dev"]}
 auto_impl = { version = "1.1", default-features = false }
 # Optional
 serde = { version = "1.0", features = ["derive", "rc"] }

